velox/connectors/hive/HiveConnector.h

```diff
@@ -92,46 +96,130 @@ class HiveTableHandle : public ConnectorTableHandle {
   const core::TypedExprPtr remainingFilter_;
 };
 
+/// Location related properties of the Hive table to be written
+class LocationHandle {
+ public:
+  enum class TableType {
+    kNew, // Write to a new table to be created.
+    kExisting, // Write to an existing table.
+    kTemporary, // Write to a temporary table.
+  };
+
+  enum class WriteMode {
+    // Write to a staging directory and then move to the target directory
+    // after write finishes.
+    kStageAndMoveToTargetDirectory,
+    // Directly write to the target directory to be created.
+    kDirectToTargetNewDirectory,
+    // Directly write to the existing target directory.
+    kDirectToTargetExistingDirectory,
+  };
+
+  LocationHandle(
+      std::string targetPath,
+      std::string writePath,
+      TableType tableType,
+      WriteMode writeMode)
+      : targetPath_(std::move(targetPath)),
+        writePath_(std::move(writePath)),
+        tableType_(tableType),
+        writeMode_(writeMode) {}
+
+  const std::string& targetPath() const {
+    return targetPath_;
+  }
+
+  const std::string& writePath() const {
+    return writePath_;
+  }
+
+  TableType tableType() const {
+    return tableType_;
+  }
+
+  WriteMode writeMode() const {
+    return writeMode_;
+  }
+
+ private:
+  // Target directory path.
+  const std::string targetPath_;
+  // Staging directory path.
+  const std::string writePath_;
+  // Whether the table to be written is new, already existing or temporary.
```

Comment:
nit: maybe move type and mode ahead of path fields?
```
// Specifies the type of table to write.

// Specifies the table write mode.
```

Reply:
Would keep it in the order of class protocol::LocationHandle